dispatcher/hub/commonHub_test.go

```diff
@@ -191,6 +191,45 @@ func TestCommonHubRun(t *testing.T) {
 	require.Nil(t, err)
 }
 
+func TestCommonHub_HandleBlockEventsBroadcast(t *testing.T) {
+	t.Parallel()
+
+	args := createMockCommonHubArgs()
+	hub, err := NewCommonHub(args)
+	require.Nil(t, err)
+
+	wg := sync.WaitGroup{}
+	numCalls := uint32(0)
+	hub.registerDispatcher(&mocks.DispatcherStub{
+		BlockEventsCalled: func(event data.BlockEvents) {
+			atomic.AddUint32(&numCalls, 1)
+			wg.Done()
+		},
+	})
+
+	hub.Subscribe(data.SubscribeEvent{
+		SubscriptionEntries: []data.SubscriptionEntry{
+			{
+				EventType: common.PushBlockEvents,
+			},
+		},
+	})
+
+	hub.Run()
+	defer hub.Close()
+	wg.Add(1)
+
+	blockEvents := data.BlockEvents{
+		Hash: "hash1",
+	}
+
+	hub.Broadcast(blockEvents)
+
+	wg.Wait()
```

Comment:
no need for this wait group.. maybe a tiny time.sleep to make sure the BlockEvents is called.. also, not sure if counter is needed in this case

Reply:
right, removed the wait group, keept the atomic counter to avoid race condition